src/opentimelineio/algo/editAlgorithm.h

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Contributors to the OpenTimelineIO project
+
+#pragma once
+
+#include "opentimelineio/composition.h"
+
+namespace opentimelineio { namespace OPENTIMELINEIO_VERSION { namespace algo {
+
+//! Enum used by 3/4 Point Edit (aka. as fill)
+enum class ReferencePoint
+{
+    Source,
+    Sequence,
+    Fit
+};
+
+// Overwrite an item or items.
+//
+// | A | B |  ->  |A| C |B|
+//   ^   ^
+//   | C |
+//
+//          item = item to overwrite (usually a clip)
```

Comment:
I think this is C in the diagram, could you make a note?

Could you write some notes about the corner cases?

C starts or ends before A
C starts or ends after B